### PR TITLE
Use scram-sha-256 password authentication for TimescaleDB

### LINKED ISSUE #1739: Use secure authentication for TimescaleDB
**Problem**
The default authentication for PostgreSQL is the insecure `md5`. Since our TimescaleDB is new we have a chance to improve the security of our database authentication without causing breaking changes.

**Solution**
Use a secure form of authentication like [SCRAM-SHA-256](https://www.postgresql.org/docs/12/auth-password.html) password authentication or if we're using TLS anyway we can utilize [certificate](https://www.postgresql.org/docs/12/auth-cert.html) authentication.

**Alternatives**

**Additional Context**

### PR DESCRIPTION
**Pull request by xin-hedera**
**Detailed description**:

- Enable scram-sha-256 in charts for timescaledb
- Enable scram-sha-256 in docker-compose.yml

**Which issue(s) this PR fixes**:
Fixes #1739 

**Special notes for your reviewer**:

- Tested docker-compose locally
- Tested charts with 1 and 2 replicas
- Didn't update timescaledb testcontainers configuration 

**Checklist**
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: charts/hedera-mirror/values.yaml [hunk 1]
```diff
@@ -193,15 +193,16 @@ timescaledb:
...
             timescaledb.max_background_workers: 20
             work_mem: 50MB
     postgresql:
       pg_hba:
-      - local     all             postgres                              peer  # allow local connections
-      - local     all             all                                   md5
-      - hostnossl all,replication all                all                md5   # allow TCP/IP connections (no ssl)
+      - local     all             postgres                            peer          # allow local connections
+      - local     all             all                                 scram-sha-256
+      - hostnossl all,replication all               all               scram-sha-256 # allow TCP/IP connections (no ssl)
```
Comment (ijungmann) [inline charts/hedera-mirror/values.yaml:205]: Add another space before comment

File: .gitignore [hunk 2]
```diff
@@ -40,6 +40,7 @@ application*.yaml
 application*.yml
 /data/
 /db/
+/timescaledb/
```
Comment (ijungmann) [inline .gitignore:43]: Is this a local directory accidentally checked in?

Comment (xin-hedera) [reply .gitignore:43]: nope. This is the host directory which mounts to the timescaledb container in docker-compose.yml. We forgot to add this to gitignore previously

Comment (ijungmann) [reply .gitignore:43]: Alright, got it!

File: charts/hedera-mirror/values.yaml (hunk 1 above)
Comment (xin-hedera) [reply charts/hedera-mirror/values.yaml:205]: sure

File: docker-compose.yml [hunk 3]
```diff
@@ -88,6 +88,9 @@ services:
     stop_signal: SIGTERM
     tty: true
     environment:
+      PGDATA: /var/lib/postgresql/data
```
Comment (xin-hedera) [inline docker-compose.yml:91]: `PGDATA` default is `/home/postgres/data`

**Review by steven-sheehy — APPROVED** [review]
LGTM

File: docker-compose.yml [hunk 4]
```diff
@@ -88,6 +88,9 @@ services:
...
+      POSTGRES_HOST_AUTH_METHOD: scram-sha-256
+      POSTGRES_INITDB_ARGS: --auth-host=scram-sha-256
```
Comment (xin-hedera) [inline docker-compose.yml:93]: this is passed to `initdb` and sets two things:

- scram-sha-256 password authentication in `pg_hba.conf` for local users via TCP/IP connections
- password_encryption = scram-sha-256 in postgresql.conf

**Review by ijungmann — APPROVED** [review]

**Review by Nana-EC — APPROVED** [review]
LGTM

**Pull request merged**